pages/task/read/index.js

```diff
@@ -21,6 +22,15 @@ module.exports = settings => {
   });
 
   app.use('/', (req, res, next) => {
+    Promise.resolve()
+      .then(() => req.user.can('profile.read.all', { establishment: req.task.data.data.establishmentId }))
+      .then(canReadAllProfiles => {
+        if (!canReadAllProfiles) {
+          const content = getContent(req.task);
+          next(new UnauthorisedError(content.errors.permissions));
+        }
+      });
```

Comment:
This seems excessively broad-scoped. It means that anyone who can't see all profiles will _always_ be thrown an error, irrespective of whether the task is related to a profile that they cant see or not.

Specifically it will prevent any basic user or ASRU user from ever being able to process any task.

I'd suggest that the permissions check, and fallback message should only be implemented in cases where an attempt to load a profile has failed.

Reply:
Leave it in the catch for now.